### ISSUE TITLE
please update SConstruct and contrib/mdx/src/SConscript

### ISSUE THREAD

**Issue by espiritocz** [isce-framework/isce2#6]
hi,

as was noticed already 2 years ago, there are some issues in 'checking for libraries/headers' when using anaconda's gdal within the isce's sconsfiguration.

I have finally *AGAIN* found the answer when installing isce on two servers and the instructions found at http://earthdef.caltech.edu/boards/4/topics/1245 helped:

in file SConstruct:

Change:
​env.PrependUnique(LIBS=['gdal'])

To:

# Resolve circular dependencies for gdal provided by Anaconda-Python...
env.PrependUnique(LIBS=['gdal', 'kea', 'hdf5', 'hdf5_hl', 'hdf5_cpp'])
# Include the following search path when linking against the conda environment, e.g. isce3.
env.AppendUnique(LINKFLAGS=['-Wl,-rpath,CONDA_ENV_LIB_DIRECTORY'])
# Include system libraries
env.AppendUnique(LINKFLAGS=['-Wl,-rpath,/usr/lib64'])

Note: Replace "CONDA_ENV_LIB_DIRECTORY" with the correct library path
(but maybe this one is not necessary?)

In file ./contrib/mdx/src/SConscript:
Change:
for i in range(envmdx['LIBS'].count('hdf5')): envmdx['LIBS'].remove('hdf5')

To:
for i in range(envmdx['LIBS'].count('hdf5')): envmdx['LIBS'].remove('hdf5')
# Remove additional libraries required by Anaconda-Python
for i in range(envmdx['LIBS'].count('kea')): envmdx['LIBS'].remove('kea')
for i in range(envmdx['LIBS'].count('hdf5_cpp')): envmdx['LIBS'].remove('hdf5_cpp')
for i in range(envmdx['LIBS'].count('hdf5_hl')): envmdx['LIBS'].remove('hdf5_hl')

**Comment by piyushrpt** [comment]
Can you issue a PR with these changes? We can take a look at it and quickly approve. 

Wouldn't adding paths like /usr/lib64 , CONDA_ENV_LIB_DIRECTORY etc to LIBPATH in SConfigISCE resolve that section?

**Comment by espiritocz** [comment]
No. That's why I write about it (and was mentioned also by the referenced user)
It seems that setting LD_LIBRARY_PATH may help but I didn't try it this time.
I read through your instructions for PR and will send soon

**Comment by piyushrpt** [comment]
Setting LD_LIBRARY_PATH / LD_RUN_PATH or activating the conda environment will help as these get set automatically then. 

One more question - is this only getting in the way when its checking for libraries, and everything works when you hit continue (as suggested in the earthdef post). For automation, we have added a --skipcheck flag to scons command that you can use. This will build the software without checking for things and will fail at the appropriate part. Looks like adding hdf5, kea etc to all the link lines but pulling them out for mdx might not be a good solution. If the changes you suggest are only for to automatically pass the tests, --skipcheck might be a better option,

**Comment by espiritocz** [comment]
thank you.
activated conda didn't help.

the check can be valuable though..
without this fix, the error message is not giving clues related to gdal (though it is visible from config.log)
instead make users search for problems in their paths to other libraries.

also I am sorry but I cannot do PR. indeed this would be my first PR (though i am not new to git).
can i ask you to check instructions from CONTRIBUTING.md, i.e. is this correct?
git remote add my_user_name https://github.com/my_user_name/isce2.git
and later:
git push my_user_name my_new_feature_branch

it says
remote: Repository not found.
fatal: repository 'https://github.com/espiritocz/isce2.git/' not found

thank you

**Comment by piyushrpt** [comment]
Ah ... You need to fork isce2 first to make sure that my_user_name/isce2.git is created before adding it as remote. Will update the instructions to reflect that.

**Comment by hfattahi** [comment]
It's actually already in the instruction:

**Initiate your work repository**
Fork isce-framework/isce from github UI, and then ....

**Comment by piyushrpt** [comment]
Looks like scons might already support RPATH - https://scons.org/doc/0.97/HTML/scons-user/a3414.html
Adding RPATH to your SConfigISCE might fix the issue, without having to add an explicit link section?

**Comment by espiritocz** [comment]
I see, i missed this part (fork from UI)

Sorry for this

po 18. 2. 2019 20:04 odesílatel piyushrpt <notifications@github.com> napsal:

**Comment by rtburns-jpl** [comment]
Can we close this issue?

**Comment by espiritocz** [comment]
I will return to it via PR once I will be compiling updated isce again...

Dne pá 22. 3. 2019 18:16 uživatel Ryan Burns <notifications@github.com>
napsal:
